gef.py

```diff
@@ -5207,6 +5205,199 @@ def do_invoke(self, args: List) -> None:
+                    _value = RIGHT_ARROW.join(dereference_from(_value))
+
+                line = f"{'  ' * depth}"
+                line += (f"{address:#x}+{_offset:#04x} {_name} : ").ljust(40)
+                line += f"{_value} ({_type.__name__})"
+                parsed_value = self.__get_ctypes_value(_structure, _name, _value)
+                if parsed_value:
+                    line += f"{RIGHT_ARROW} {parsed_value}"
+                gef_print(line)
+
+                if issubclass(_type, ctypes.Structure):
+                    self.apply_at(address + _offset, depth + 1, max_depth)
+                elif _type.__name__.startswith("LP_"): # hack
+                    __sub_type_name = _type.__name__.replace("LP_", "")
+                    __deref = u64(gef.memory.read(address + _offset, 8))
+                    self.apply_at(__deref, depth + 1, max_depth)
+            return
+
+        def __get_ctypes_value(self, struct, item, value) -> str:
+            if not hasattr(struct, "_values_"): return ""
+            values_list = getattr(struct, "_values_")
+            default = ""
+            for name, values in values_list:
+                if name != item: continue
+                if callable(values):
+                    return values(value)
+                try:
+                    for val, desc in values:
+                        if value == val: return desc
+                        if val is None: default = desc
+                except:
+                    err(f"Error while trying to obtain values from _values_[\"{name}\"]")
+            return default
+
+    class Module:
+        def __init__(self, path: pathlib.Path) -> None:
+            self.path = path
+            self.raw = self.__load()
+            self.structures : List[ExternalStructureManager.Structure] = []
+
+            for entry in self:
+                self.structures.append(ExternalStructureManager.Structure(self.path, entry))
+            return
+
+        def __load(self) -> ModuleType:
+            """Load a custom module, and return it"""
+            fpath = self.path
+            spec = importlib.util.spec_from_file_location(fpath.stem, fpath)
+            module = importlib.util.module_from_spec(spec)
+            sys.modules[fpath.stem] = module
+            spec.loader.exec_module(module)
+            return module
+
+        def __str__(self) -> str:
+            return self.path.stem
+
+        def __iter__(self) -> Generator[str, None, None]:
+            _invalid = {"BigEndianStructure", "LittleEndianStructure", "Structure"}
+            _structs = {x for x in dir(self.raw) \
+                             if inspect.isclass(getattr(self.raw, x)) \
+                             and issubclass(getattr(self.raw, x), ctypes.Structure)}
+            for entry in (_structs - _invalid):
+                yield entry
+            return
+
+        def __contains__(self, structure_name: str) -> bool:
+            for structure in self.structures:
```

Comment:
Maybe we should makes `structures` a `Dict` of `name: Structure`. It would make `__contains__` easier and more efficient:

```python
return structure_name in self.structures
```